SixTrack/tools/astuce/astuce++.cpp

```diff
@@ -0,0 +1,708 @@
+#include "astuce++.h"
+#include <cctype>
+#include <iomanip>
+#include <cstdlib>
+
+void Astuce::Run()
+{
+	LoadInputMask();
+	ReadInputMaskFile();
+	if(OutputFlag){PrintInputSettings();}
+	ReadInputFiles();
+	ParseSourceFile();
+	DoOutput();
+}
+
+void Astuce::LoadInputMask()
+{
+	std::ifstream *infile = new std::ifstream(InputFileName);
+
+	if(infile->good())
+	{
+		if(OutputFlag)
+		{
+			*LogStream << "Opened " << InputFileName << std::endl;
+		}
+	}
+	else
+	{
+		std::cerr << "Could not open " << InputFileName << std::endl;
+		exit(EXIT_FAILURE);
+	}
+
+	char c_buffer[4096];
+	while(infile->good())
+	{
+		infile->getline(c_buffer, 4096);
+		if(!infile->eof())
+		{
+			InputBuffer.push_back(c_buffer);
+		}
+	}
+
+	infile->close();
+	delete infile;
+}
+
+void Astuce::ReadInputMaskFile()
+{
+	size_t line = 1;
+	std::string Buffer;
+
+	std::vector<std::string>::const_iterator itr = InputBuffer.begin();
+	while(itr != InputBuffer.end())
+	{
+		if(line == 1)
+		{
+			Buffer = *itr;
+
+			size_t space = 0;
+			size_t old_pos = 0;
+			while(space!=std::string::npos)
+			{
+				space = Buffer.find(" ", space+1);
+				InputFileNames.insert(Buffer.substr(old_pos,space-old_pos));
+				old_pos = space+1;
+			}
+		}
+		else if(line == 2)
+		{
+			OutputFileName = *itr;
+		}
+		else
+		{
+			if(itr->substr(0,2) == "EX" || itr->substr(0,2) == "ex")
+			{
+				//Exit processing
+				break;
+			}
+			else if(itr->substr(0,2) == "DF" || itr->substr(0,2) == "df")
+			{
+				//flag defines
+				//Need to extract first, then split on a ',' char
+				size_t start = itr->find_first_not_of(" ", 2);
+				if(start!=std::string::npos)
+				{
+					std::string defines = itr->substr(start,itr->size()-start);
+					size_t step = 0;
+					size_t old_step = 0;
+					while(step!=std::string::npos)
+					{
+						step = defines.find(",", step+1);
+						Flags.insert(FixCase(defines.substr(old_step,step-old_step)));
+						old_step = step+1;
+					}
+				}
+			}
+			else if(itr->substr(0,2) == "E " || itr->substr(0,2) == "e ")
```

Comment:
Here you look for "E ", higher up you look for "EX" and "DF". Can there be lines "DF" without a space before the actual flags?